runtime/lua/vim/_extui.lua

```diff
@@ -0,0 +1,106 @@
+local api = vim.api
+local ext = require('vim._extui.shared')
+ext.msg = require('vim._extui.messages')
+ext.cmd = require('vim._extui.cmdline')
+local M = {}
+
+local function ui_callback(event, ...)
+  local handler = ext.msg[event] or ext.cmd[event]
+  if not handler then
+    return
+  end
+  ext.tab_check_wins()
+  handler(...)
+  api.nvim__redraw({
+    flush = true,
+    cursor = handler == ext.cmd[event] and true or nil,
+    win = handler == ext.cmd[event] and ext.wins[ext.tab].cmd or nil,
+  })
+end
+local scheduled_ui_callback = vim.schedule_wrap(ui_callback)
+
+M.enable = function(opts)
+  vim.validate('opts', opts, 'table', true)
+  ext.cfg = vim.tbl_deep_extend('keep', opts, ext.cfg)
+  if ext.cfg.enable == false then
+    -- Detach and cleanup windows, buffers and autocommands.
+    for _, tab in ipairs(api.nvim_list_tabpages()) do
+      for _, win in pairs(ext.wins[tab] or {}) do
+        api.nvim_win_close(win, true)
+      end
+    end
+    for _, buf in pairs(ext.bufs) do
+      api.nvim_buf_delete(buf, {})
+    end
+    api.nvim_clear_autocmds({ group = ext.augroup })
+    vim.ui_detach(ext.ns)
+    return
+  end
+
+  api.nvim_set_hl(ext.ns, 'Normal', { link = 'MsgArea' })
```

Comment:
I tried to link `Normal` differently in the history and the message window (to dim only disappearing messages) but currently it seems to be not simple from the user side (at first glance it requires `=require('vim._extui.shared').wins[tabnr].box` and set `winhighlight`) or I have overlooked something.

Reply:
Yes a new highlight group to apply to the message box is probably in order. Also not yet applying `MoreMsg` to the new more window.